### PR TITLE
action: convert and check top images for zran

### PR DESCRIPTION
**Pull request by imeoer**
To ensure zran images work fine on conversion and run.

Example: https://github.com/imeoer/image-service/actions/runs/4061749342

Signed-off-by: Yan Song <imeoer@linux.alibaba.com>

### CODE REVIEW

File: .github/workflows/convert.yml [hunk 1]
```diff
@@ -14,34 +14,143 @@ env:
...
+                 --fs-version 5
+
+            sudo rm -rf ./tmp
+            sudo DOCKER_CONFIG=$HOME/.docker nydusify check --source $I:latest \
+                --target localhost:5000/$I:nydus-nightly-oci-ref
+          done
+
+  convert-native-v5:
+    runs-on: ubuntu-latest
+    needs: [nydusify-build, nydus-build, fsck-erofs-build]
```
Comment (bergwolf) [inline .github/workflows/convert.yml:131]: This does not really depend on fsck-erofs-build, right?

File: .github/workflows/convert.yml [hunk 2]
```diff
@@ -60,8 +169,39 @@ jobs:
             sudo DOCKER_CONFIG=$HOME/.docker nydusify check --source $I:latest \
                 --target localhost:5000/$I:nydus-nightly-v5
           done
+
+  convert-native-v6:
+    runs-on: ubuntu-latest
+    needs: [nydusify-build, nydus-build, fsck-erofs-build]
```
Comment (bergwolf) [inline .github/workflows/convert.yml:175]: same here.

File: .github/workflows/convert.yml [hunk 3]
```diff
@@ -60,8 +169,39 @@ jobs:
...
+          path: /usr/local/bin
+      - name: Download fsck.erofs
+        uses: actions/download-artifact@master
+        with:
+          name: fsck-erofs-artifact
+          path: /usr/local/bin
       - name: Convert and check RAFS v6 images
         run: |
+          sudo chmod +x /usr/local/bin/nydus*
+          sudo chmod +x /usr/local/bin/fsck.erofs
```
Comment (bergwolf) [inline .github/workflows/convert.yml:203]: Is it needed?

File: .github/workflows/convert.yml (hunk 1 above)
Comment (imeoer) [reply .github/workflows/convert.yml]: Fixed

File: .github/workflows/convert.yml (hunk 2 above)
Comment (imeoer) [reply .github/workflows/convert.yml:175]: Fixed

File: .github/workflows/convert.yml (hunk 3 above)
Comment (imeoer) [reply .github/workflows/convert.yml:203]: It seems yes, `actions/upload-artifact` lost x permission.

**Comment by imeoer** [comment]
https://github.com/imeoer/image-service/actions/runs/4053624546/jobs/6974507870

Action hanging in zran image conversion job, will take a look before merging this. cc @bergwolf

File: .github/workflows/convert.yml [hunk 4]
```diff
@@ -14,34 +14,143 @@ env:
...
+          sudo chmod +x /usr/local/bin/nydus*
+          sudo docker run -d --restart=always -p 5000:5000 registry
+          for I in $(cat ${{ env.IMAGE_LIST_PATH }}); do
+            echo "converting $I:latest to $I:nydus-nightly-oci-ref"
+            # for pre-built images
+            sudo DOCKER_CONFIG=$HOME/.docker nydusify convert \
+                 --oci-ref \
+                 --source $I:latest \
+                 --target ${{ env.REGISTRY }}/${{ env.ORGANIZATION }}/$I:nydus-nightly-oci-ref \
+                 --fs-version 5
```
Comment (jiangliu) [inline .github/workflows/convert.yml:117]: Should we use v6 for zran?

**Comment by jiangliu** [comment]
Fixes #1034

File: .github/workflows/convert.yml (hunk 4 above)
Comment (imeoer) [reply .github/workflows/convert.yml]: Thanks, fixed, we still enforce to use v6 for `--oci-ref` image.

**Review by jiangliu — APPROVED** [review]

**Pull request merged**